src/main/scala/mesosphere/marathon/state/Instance.scala

```diff
@@ -41,7 +42,10 @@ case class Instance(
     * @param runSpec The run spec belonging to this instance.
     * @return This instance in core model.
     */
-  def toCoreInstance(runSpec: RunSpec) = CoreInstance(instanceId, agentInfo, state, tasksMap, runSpec, reservation)
+  def toCoreInstance(runSpec: RunSpec) = {
+    require(role.isDefined, s"BUG! Stored $instanceId has no role defined which should be covered by migration.")
```

Comment:
Yes. Role should be non-optional. Thank you.

Reply:
I tend to disagree. This is not internal `core.instance.Instance` but storage `state.Instance` model. It is used for migrations and loading only and is similar to `StoredGroup`. I think it's a valid pattern. We could start cleaning up the different storage models JSON, Protobuf, RAML, `state.Instance` and `storage.StoredGroup` and just have one. However, if we eg would go with Protobuf it would also just have the role as an optional field if I'm not mistaken.

Reply:
We have protobuf, then we have this, then we have internal state?

Why not migrate it directly on the protobuf?

Reply:
I'd prefer to not rely on secondary models for migration since they could evolve in backwards incompatible ways

Reply:
🤦‍♂ issue is we don't use protobuf for instances. While I'm inclined to think that we should, that's orthogonal to this change.